export/export.py

```diff
@@ -0,0 +1,866 @@
+            export_recipe: Contains the configuration for the export process
+            name: Optional name for the export
+            dynamic_shapes: Optional dynamic shape specifications
+            constant_methods: Optional dictionary of constant methods
+            artifact_dir: Optional directory to store artifacts
+        """
+        # Standardize model to dictionary format
+        self._model = model if isinstance(model, dict) else {"forward": model}
+
+        # Standardize example_inputs to dictionary format
+        self._example_inputs = (
+            example_inputs
+            if isinstance(example_inputs, dict)
+            else {"forward": example_inputs}
+        )
+
+        # Standardize dynamic_shapes to dictionary format
+        self._dynamic_shapes = {}
+        if dynamic_shapes is not None:
+            if isinstance(dynamic_shapes, dict):
+                self._dynamic_shapes = dynamic_shapes
+            else:
+                self._dynamic_shapes = {"forward": dynamic_shapes}
+
+        self._name = name
+        self._constant_methods = constant_methods
+        self._artifact_dir = artifact_dir
+        self._export_recipe = export_recipe
+
+        # Initialize stages
+        self._stages = {}
+
+        # Create the source transform stage if a quantization recipe is provided
+        if self._export_recipe.quantization_recipe is not None:
+            self._stages["source_transform"] = SourceTransformStage(
+                quantization_recipe=self._export_recipe.quantization_recipe
+            )
+
+            # Create the quantize stage if a quantizer is provided
+            if self._export_recipe.quantization_recipe is not None:
+                quantizer = self._export_recipe.quantization_recipe.get_quantizer()
+                if quantizer is not None:
+                    self._stages["quantize"] = QuantizeStage(quantizer=quantizer)
+
+        # Create the export stage
+        self._stages["export"] = ExportStage(
+            pre_edge_transform_passes=self._export_recipe.pre_edge_transform_passes
+        )
+
+        # Create the edge transform and lower stage
+        self._stages["edge_transform_and_lower"] = EdgeTransformAndLowerStage(
+            partitioners=self._export_recipe.partitioners,
+            transform_passes=self._export_recipe.edge_transform_passes,
+            compile_config=self._export_recipe.edge_compile_config,
+        )
+
+        # Create the executorch stage
+        self._stages["executorch"] = ExecutorchStage(
+            backend_config=self._export_recipe.executorch_backend_config
+        )
+
+        # Define the pipeline as a list of stages in execution order
+        self._pipeline = ["export", "edge_transform_and_lower", "executorch"]
+
+        # If quantization is available, add it after source transform
+        if "quantize" in self._stages:
+            self._pipeline.insert(0, "quantize")
```

Comment:
As I said elsewhere I think quantize should not be responsible for running export. We should run export, run quantizer, prepare, convert, and run export again. This call can be managed within here by detecting the need to run export again if we have pt2e quantize requested

Reply:
We need to store this because it's used in all the later stages such as to_edge etc.